Framework/PythonInterface/mantid/kernel/src/Environment/ErrorHandling.cpp

```diff
@@ -64,8 +70,7 @@ void throwRuntimeError(const bool withTrace) {
   Py_XDECREF(traceback);
   Py_XDECREF(exception);
   Py_XDECREF(value);
-  // Raise this error as a C++ error
-  throw std::runtime_error(msg.str());
+  m_msg = builder.str();
```

Comment:
Thanks for pointing this out. I needed `noexcept` for `what()` but i didn't think enough for the constructor. The string allocation could throw so I will remove the `noexcept` marker